src/main/java/graphql/validation/rules/DeferDirectiveOnRootLevel.java

```diff
@@ -0,0 +1,57 @@
+package graphql.validation.rules;
+
+import graphql.language.Directive;
+import graphql.language.FragmentDefinition;
+import graphql.language.InlineFragment;
+import graphql.language.Node;
+import graphql.language.OperationDefinition;
+import graphql.language.SelectionSet;
+import graphql.validation.AbstractRule;
+import graphql.validation.ValidationContext;
+import graphql.validation.ValidationErrorCollector;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static graphql.validation.ValidationErrorType.MisplacedDirective;
+
+public class DeferDirectiveOnRootLevel extends AbstractRule {
+
+    public DeferDirectiveOnRootLevel(ValidationContext validationContext, ValidationErrorCollector validationErrorCollector) {
+        super(validationContext, validationErrorCollector);
+        this.setVisitFragmentSpreads(true);
+    }
+
+    @Override
+    public void checkDirective(Directive directive, List<Node> ancestors) {
+        if (directive.getName().equals("defer")){
+            Optional<Node> fragmentAncestor = getFragmentAncestor(ancestors);
+            if (fragmentAncestor.isPresent() && fragmentAncestor.get() instanceof OperationDefinition){
+                OperationDefinition operationDefinition = (OperationDefinition) fragmentAncestor.get();
+                if (OperationDefinition.Operation.MUTATION.equals(operationDefinition.getOperation()) || OperationDefinition.Operation.SUBSCRIPTION.equals(operationDefinition.getOperation())) {
+                    String message = i18n(MisplacedDirective, "DirectiveMisplaced.notAllowedOperationRootLevel", directive.getName(), operationDefinition.getOperation().name().toLowerCase());
+                    addError(MisplacedDirective, directive.getSourceLocation(), message);
```

Comment:
This error enum seems fair enough

Reply:
Hang on - this code looks nearly the same as the one rule below

What does `DeferDirectiveOnRootLevel` mean?

Can we link to the spec per rule which allows one know what is being enforced

Reply:
https://github.com/graphql/graphql-spec/pull/742">spec/Section 5 -- Validation.md ### Defer And Stream Directives Are Used On Valid Operations
(i've added on class javadoc